### PR TITLE
use the values from matrix.json in JS tests

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: .github/workflows/js_tests.yml [hunk 1]
```diff
@@ -12,27 +12,41 @@ permissions:
...
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        run: |
+          {
+            echo 'matrix<<EOF'
+            curl --silent --show-error --fail https://raw.githubusercontent.com/theforeman/foreman/${{ github.base_ref }}/.github/matrix.json
```
Comment (ekohl) [inline .github/workflows/js_tests.yml]: This means it won't work for forks. I'd prefer to use `actions/checkout` to get the real file.

**Review by ekohl — CHANGES_REQUESTED** [review]

File: .github/workflows/js_tests.yml (hunk 1 above)
Comment (ekohl) [reply .github/workflows/js_tests.yml]: Or you can use the correct namespace and repo name from variables.
```suggestion
            curl --silent --show-error --fail https://raw.githubusercontent.com/${{ github.repository }}/${{ github.base_ref }}/.github/matrix.json
```

Comment (evgeni) [reply .github/workflows/js_tests.yml]: eh? `base_ref` is the PR target, which I just assume to be `theforeman/foreman`.

or do you mean PRs against forks? :D

Comment (ekohl) [reply .github/workflows/js_tests.yml]: Hmm no. What I mean is that you should be able to create a PR that changes Ruby/Node and CI should respect that.

So what I'd expect is that you have the equivalent of `actions/checkout` and take `.github/matrix.json` from that.

Comment (evgeni) [reply .github/workflows/js_tests.yml]: Mhh, OK.

Comment (ekohl) [reply .github/workflows/js_tests.yml]: But you're right that `base_ref` is not right. I think it should be `ref` or `ref_name`.

**Comment by evgeni** [comment]
see https://github.com/theforeman/foreman/pull/9881#pullrequestreview-1705544985 and https://github.com/ekohl/gha-matrix-builder/pull/1

File: .github/workflows/js_tests.yml [hunk 2]
```diff
@@ -12,27 +12,38 @@ permissions:
...
       - name: Checkout Foreman
         uses: actions/checkout@v4
       - uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 2.7
-      - name: Setup Node
-        uses: actions/setup-node@v4
+          ruby-version: ${{ matrix.ruby }}
+      - name: Setup Node ${{ matrix.node }}
+        uses: actions/setup-node@v3
```
Comment (ekohl) [inline .github/workflows/js_tests.yml]:
```suggestion
        uses: actions/setup-node@v4
```

File: .github/workflows/js_tests.yml [hunk 3]
```diff
@@ -12,27 +12,38 @@ permissions:
...
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: ekohl/gha-matrix-builder@v0
+        with:
+          base_matrix_url: "https://raw.githubusercontent.com/${{ github.repository }}/${{ github.base_ref }}/.github/matrix.json"
```
Comment (ekohl) [inline .github/workflows/js_tests.yml]:
```suggestion
        uses: theforeman/gha-matrix-builder@v0
```

File: .github/workflows/js_tests.yml [hunk 4]
```diff
@@ -12,27 +12,38 @@ permissions:
...
       contents: read  # for actions/checkout to fetch code
     runs-on: ubuntu-latest
+    needs: setup_matrix
     timeout-minutes: 20
     strategy:
       fail-fast: false
-      matrix:
-        node-version:
-          - 14
+      matrix: ${{ fromJson(needs.setup_matrix.outputs.matrix) }}
```
Comment (ekohl) [inline .github/workflows/js_tests.yml:34]: I wonder if we can somehow be smart and only test on the first Ruby version. Running the JS tests with all Ruby versions probably doesn't make sense.
```suggestion
      matrix:
        ruby:
          - ${{ fromJson(needs.setup_matrix.outputs.matrix).ruby[0] }}
        node: ${{ fromJson(needs.setup_matrix.outputs.matrix).node }}
```

But perhaps with webpack 5 the webpack config no longer executes Ruby to determine plugins (not sure, couldn't see it immediately) and it no longer needs Ruby.

**Review by ekohl — APPROVED** [review]

**Pull request merged**